### ISSUE TITLE
Nested timestamps cause a DataException to be thrown by kafka sink connector

### ISSUE THREAD

**Issue by c4181** [neo4j-contrib/neo4j-streams#580]
## Expected Behavior (Mandatory)
Events with a timestamp nested in a struct should be parseable in the connector

## Actual Behavior (Mandatory)
Events with a timestamp nested in a struct do not process properly

```
streams.service.errors.ProcessingError: Error processing 6 messages
ErrorData(originalTopic=neotopic, timestamp=-1, partition=1, offset=42, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)
ErrorData(originalTopic=neotopic, timestamp=-1, partition=1, offset=42, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)
ErrorData(originalTopic=neotopic, timestamp=-1, partition=1, offset=42, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)
ErrorData(originalTopic=neotopic, timestamp=-1, partition=1, offset=42, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)
ErrorData(originalTopic=neotopic, timestamp=-1, partition=1, offset=42, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)
ErrorData(originalTopic=foo, timestamp=-1, partition=1, offset=43, exception=org.apache.kafka.connect.errors.DataException: Exception thrown while processing field 'place', key=null, value=Struct{firstName=Alex,lastName=Smith,age=21,bool=true,short=1234,byte=-32,long=12425436,float=2356.3,double=-2436546.56457,modified=Fri Sep 23 16:10:02 EDT 2016,place=Struct{name=San Mateo (CA),latitu, executingClass=class streams.kafka.connect.sink.Neo4jSinkTask)

	at streams.service.errors.KafkaErrorService.report(KafkaErrorService.kt:36)
	at streams.kafka.connect.sink.Neo4jSinkTask.put(Neo4jSinkTask.kt:45)
	at streams.kafka.connect.sink.Neo4jSinkTaskTest.should insert data into Neo4j(Neo4jSinkTaskTest.kt:183)
```

## How to Reproduce the Problem
Sink an event to Neo4j with a timestamp nested inside of a struct

### Simple Dataset (where it's possible)
I've adapted the existing schema/value set in the Neo4jSinkTaskTest to illustrate the problem.
```kotlin
    private val PLACE_SCHEMA = SchemaBuilder.struct().name("com.example.Place")
            .field("name", Schema.STRING_SCHEMA)
            .field("latitude", Schema.FLOAT32_SCHEMA)
            .field("longitude", Schema.FLOAT32_SCHEMA)
            .field("modified", Timestamp.SCHEMA)
            .build()

    private val PERSON_SCHEMA = SchemaBuilder.struct().name("com.example.Person")
            .field("firstName", Schema.STRING_SCHEMA)
            .field("lastName", Schema.STRING_SCHEMA)
            .field("age", Schema.OPTIONAL_INT32_SCHEMA)
            .field("bool", Schema.OPTIONAL_BOOLEAN_SCHEMA)
            .field("short", Schema.OPTIONAL_INT16_SCHEMA)
            .field("byte", Schema.OPTIONAL_INT8_SCHEMA)
            .field("long", Schema.OPTIONAL_INT64_SCHEMA)
            .field("float", Schema.OPTIONAL_FLOAT32_SCHEMA)
            .field("double", Schema.OPTIONAL_FLOAT64_SCHEMA)
            .field("modified", Timestamp.SCHEMA)
            .field("place", PLACE_SCHEMA)
            .build()

    val place = Struct(PLACE_SCHEMA)
            .field("name", Schema.STRING_SCHEMA)
            .field("latitude", Schema.FLOAT32_SCHEMA)
            .field("longitude", Schema.FLOAT32_SCHEMA)
            .put("modified", Date(1474661402123L))

    val struct= Struct(PERSON_SCHEMA)
            .put("firstName", "Alex")
            .put("lastName", "Smith")
            .put("bool", true)
            .put("short", 1234.toShort())
            .put("byte", (-32).toByte())
            .put("long", 12425436L)
            .put("float", 2356.3.toFloat())
            .put("double", -2436546.56457)
            .put("age", 21)
            .put("modified", Date(1474661402123L))
            .put("place", place)
```

Sample Cypher
```
CREATE (n:PersonExt {name: event.firstName, surname: event.lastName})
```

### Steps (Mandatory)

  1. Send event to Kafka topic
  2. Sink task fails when parsing event

Currently used versions

### Versions

  - OS: RHEL 8.7
  - Neo4j: 4.4.6
  - Neo4j-Streams: 5.0.2

**Comment by c4181** [comment]
I believe this is caused by calling setMap at the end of the setStructField method in Neo4JValueConverter. At the end of setStructField, converted is a map that contains the timestamp as a LocalDateTime. When calling setMap, it expects timestamps to be a Date.

Changing setMap to setValue solved the issue I'm experiencing and existing unit tests still pass.

https://github.com/neo4j-contrib/neo4j-streams/blob/6ab213ee46d9d29682009bc3e8060956bc5f6f08/kafka-connect-neo4j/src/main/kotlin/streams/kafka/connect/sink/converters/Neo4jValueConverter.kt#L57-L62

**Comment by c4181** [comment]
Fixed by #582
